update_sample_status/main.py

```diff
@@ -43,26 +51,60 @@ def update_airtable(project_config: dict, sample: str, status: str):
     return ('', 204)
 
 
-def update_sm(project: str, sample: str, status: str, batch: str):
-    """Update the status of a sequence given it's corresponding
-    external sample id"""
+def update_sm(project: str, external_sample_id: str, status: str, batch: str):
+    """Update the status of a sequence given it's corresponding external sample id"""
 
-    seqapi = SequenceApi()
+    samapi = SampleApi()
     sequence_type = SequenceType('genome')
     sample_type = SampleType('blood')
     seq_meta = {'batch': batch}
+    external_sequence_id: Optional[str] = None
 
-    try:
-        sequence_model = SequenceUpdateModel(
-            status=SequenceStatus(status), meta=seq_meta
+    sid_map = samapi.get_sample_id_map_by_internal([external_sample_id])
+    if sid_map and external_sample_id in sid_map:
+        internal_sample_id = sid_map[external_sample_id]
+    else:
+        samapi.create_new_sample(
+            project=project,
+            new_sample=NewSample(
+                external_id=external_sample_id,
+                type=sample_type,
+            ),
         )
-        seqapi.upsert_sequence_from_external_sample_and_type(
-            external_sample_id=sample,
-            sequence_type=sequence_type,
-            sequence_update_model=sequence_model,
+
+        # should create here if it doesn't exist
+        return abort(
```

Comment:
I'm not sure that aborting is what we want to do at this stage? Previously with this function, if the sample didn't exist it would create it and proceed with creating the sequence. Is there a reason we abort and exit here?